mmengine/visualization/vis_backend.py

```diff
@@ -816,3 +816,142 @@ def _flatten(self, d, parent_key='', sep='.') -> dict:
+
+        task_kwargs = \
+            self._init_kwargs if self._init_kwargs is not None else {}
+        self._clearml = clearml
+        self._task = self._clearml.Task.init(**task_kwargs)
+        self._logger = self._task.get_logger()
+
+    @property  # type: ignore
+    @force_init_env
+    def experiment(self):
+        """Return clearml object."""
+        return self._clearml
+
+    @force_init_env
+    def add_config(self, config: Config, **kwargs) -> None:
+        """Record the config to clearml.
+
+        Args:
+            config (Config): The Config object
+        """
+        self._cfg = config
+        self._task.connect_configuration(vars(config))
+
+    @force_init_env
+    def add_image(self,
+                  name: str,
+                  image: np.ndarray,
+                  step: int = 0,
+                  **kwargs) -> None:
+        """Record the image to clearml.
+
+        Args:
+            name (str): The image identifier.
+            image (np.ndarray): The image to be saved. The format
+                should be RGB.
+            step (int): Global step value to record. Default to 0.
+        """
+        self._logger.report_image(
+            title=name, series=name, iteration=step, image=image)
+
+    @force_init_env
+    def add_scalar(self,
+                   name: str,
+                   value: Union[int, float, torch.Tensor, np.ndarray],
+                   step: int = 0,
+                   **kwargs) -> None:
+        """Record the scalar data to clearml.
+
+        Args:
+            name (str): The scalar identifier.
+            value (int, float, torch.Tensor, np.ndarray): Value to save.
+            step (int): Global step value to record. Default to 0.
+        """
+        self._logger.report_scalar(
+            title=name, series=name, value=value, iteration=step)
+
+    @force_init_env
+    def add_scalars(self,
+                    scalar_dict: dict,
+                    step: int = 0,
+                    file_path: Optional[str] = None,
+                    **kwargs) -> None:
+        """Record the scalar's data to clearml.
+
+        Args:
+            scalar_dict (dict): Key-value pair storing the tag and
+                corresponding values.
+            step (int): Global step value to record. Default to 0.
+            file_path (str, optional): Useless parameter. Just for
+                interface unification. Default to None.
+        """
+        assert 'step' not in scalar_dict, 'Please set it directly ' \
+                                          'through the step parameter'
+        for key, value in scalar_dict.items():
+            self._logger.report_scalar(
+                title=key, series=key, value=value, iteration=step)
+
+    def close(self) -> None:
+        """Close the clearml."""
+        file_paths: List[str] = list()
+        for filename in scandir(self._cfg.work_dir, self._artifact_suffix,
```

Comment:
If `add_config` is not called, an error will be rasied here.
```suggestion
        for filename in scandir(self._save_dir, self._artifact_suffix,
```

Reply:
Since all trained model, configuration are stored in  self._cfg.work_dir . 
I think using self._cfg.work_dir is proper here/ 
work_dirs/
└── resnet18_8xb16_cifar10_clearml     :   <-----  self._cfg.work_dir
    ├── 20230509_135231                        :  <-----  self._save_dir
    ├── epoch_1.pth
    ├── last_checkpoint
    └── resnet18_8xb16_cifar10_clearml.py

Clearml doesn't save anything inside self._save_dir. 
So if this is self._save_dir, we couldn't find any desired files.

Reply:
Error catch added